src/service/resources-impl.js

```diff
@@ -239,43 +278,147 @@ export class ResourcesImpl {
     }
   }
 
+  /**
+   * @param {!Array<!IntersectionObserverEntry>} entries
+   * @param {!IntersectionObserver} unusedObserver
+   * @private
+   */
+  intersects_(entries, unusedObserver) {
+    dev().fine(TAG_, 'intersect', entries);
+
+    const toUnload = [];
+
+    const promises = entries.map(entry => {
+      const {boundingClientRect, isIntersecting, target, rootBounds} = entry;
+      // Closure Compiler doesn't recognize boundingClientRect as a ClientRect.
+      const clientRect = /** @type {!ClientRect} */ (boundingClientRect);
```

Comment:
Ditto: do we actually have a `ClientRect`? There might be a good reason Closure doesn't recognize it :)